apollo-router/src/plugins/connectors/make_requests.rs

```diff
@@ -92,6 +97,54 @@ impl RequestInputs {
             }
         }
 
+        // Add headers from the original router request.
+        // Only include headers that are actually referenced to save on passing around unused headers in memory.
+        if variables_used.contains(&Namespace::Request) {
+            let headers: Map<ByteString, Value> = supergraph_request
+                .headers()
+                .iter()
+                .filter_map(|(key, value)| {
+                    if headers_used.contains::<String>(&key.as_str().into()) {
```

Comment:
Cloning the key here partially defeats the purpose. Instead:
```suggestion
                    if headers_used.contains(key.as_str()) {
```

Reply:
I could have sworn this did not work before... I dunno what I did before but this works fine now! lol 😓